app/scripts/app.js

```diff
@@ -157,56 +157,64 @@ blocJams.controller('PlayerBar.controller', ['$scope', 'SongPlayer', function($s
        formats: [ "mp3" ],
        preload: true
     });
- 
+
        this.play();
      }
    };
  });
-blocJams.directive('slider',function(){
-   var updateSeekPercentage = function($seekBar, event) {
-     var barWidth = $seekBar.width();
-     var offsetX =  event.pageX - $seekBar.offset().left;
- 
-     var offsetXPercent = (offsetX  / $seekBar.width()) * 100;
+blocJams.directive('slider', ['$document', function($document){
+  var calculateSliderPercentFromMouseEvent = function($slider, event) {
+     var offsetX =  event.pageX - $slider.offset().left; // Distance from left
+     var sliderWidth = $slider.width(); // Width of slider
+     var offsetXPercent = (offsetX  / sliderWidth);
      offsetXPercent = Math.max(0, offsetXPercent);
-     offsetXPercent = Math.min(100, offsetXPercent);
- 
-     var percentageString = offsetXPercent + '%';
-     $seekBar.find('.fill').width(percentageString);
-     $seekBar.find('.thumb').css({left: percentageString});
-   }
-   
-  
-  return{
+     offsetXPercent = Math.min(1, offsetXPercent);
+     return offsetXPercent;
+   };
+   return{
     templateUrl: '/templates/directives/slider.html', // We'll create these files shortly.
     replace: true,
     restrict: 'E',
+    scope: {},
     link: function(scope, element, attributes) {
- 
+      scope.value = 0;
+      scope.max = 200;
       var $seekBar = $(element);
- 
-      $seekBar.click(function(event) {
-        updateSeekPercentage($seekBar, event);
-      });
- 
-      $seekBar.find('.thumb').mousedown(function(event){
-        $seekBar.addClass('no-animate');
- 
-        $(document).bind('mousemove.thumb', function(event){
-          updateSeekPercentage($seekBar, event);
-        });
- 
-        //cleanup
-        $(document).bind('mouseup.thumb', function(){
-          $seekBar.removeClass('no-animate');
-          $(document).unbind('mousemove.thumb');
-          $(document).unbind('mouseup.thumb');
-        });
- 
-      });
+      var percentString = function () {
+      var percent = Number(scope.value) / Number(scope.max) * 100;
+        return percent + "%";
+      };
+
+      scope.fillStyle = function() {
+        return {width: percentString()};
+      };
+
+      scope.thumbStyle = function() {
+        return {left: percentString()};
+      };
+
+      scope.onClickSlider = function(event) {
+        var percent = calculateSliderPercentFromMouseEvent($seekBar, event);
+        scope.value = percent * scope.max;
+      };
+
+      scope.trackThumb = function() {
```

Comment:
Aaaaah I feel so dumb! Must be more careful and thorough. Thank you.

On Tue, Aug 11, 2015 at 10:16 AM, Phillip Spitler <notifications@github.com>
wrote:

> In app/scripts/app.js
> <https://github.com/lukkyjoe/bloc-jams/pull/4#discussion_r36757077>:
>
> > +      };
> > +
> > +      scope.fillStyle = function() {
> > +        return {width: percentString()};
> > +      };
> > +
> > +      scope.thumbStyle = function() {
> > +        return {left: percentString()};
> > +      };
> > +
> > +      scope.onClickSlider = function(event) {
> > +        var percent = calculateSliderPercentFromMouseEvent($seekBar, event);
> > +        scope.value = percent * scope.max;
> > +      };
> > +
> > +      scope.trackThumb = function() {
>
> I don't see this function being defined in your code
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lukkyjoe/bloc-jams/pull/4/files#r36757077>.
>